clang/lib/CIR/CodeGen/CIRGenItaniumCXXABI.cpp

```diff
@@ -2288,6 +2288,23 @@ static mlir::cir::FuncOp getItaniumDynamicCastFn(CIRGenFunction &CGF) {
   return CGF.CGM.createRuntimeFunction(FTy, "__dynamic_cast");
 }
 
+mlir::cir::DynamicCastToVoidInfoAttr
+CIRGenItaniumCXXABI::buildDynamicCastToVoidInfo(CIRGenFunction &CGF,
+                                                QualType SrcRecordTy) {
+  mlir::Type vtableElemTy;
+  uint64_t vtableElemAlign;
+  if (CGM.getItaniumVTableContext().isRelativeLayout()) {
+    vtableElemTy = CGF.getBuilder().getSInt32Ty();
+    vtableElemAlign = 4;
+  } else {
+    vtableElemTy = CGF.ConvertType(CGF.getContext().getPointerDiffType());
```

Comment:
This is an AST query, which is also availble in LoweringPrepare.

Reply:
Updated.